packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -217,84 +324,67 @@ exports.createPages = async ({ graphql, actions, reporter }, pluginOptions) => {
   if (navigationYamlResult.errors) {
     reporter.panicOnBuild('🚨  ERROR: Loading "allNavigationYaml" query');
   }
-  const pages = allContentPagesResult.data.contents.nodes;
+  const pages = result.data.allContentPage.nodes;
   const navigationPages = navigationYamlResult.data.allNavigationYaml.nodes.reduce(
     (pageLinks, node) => [...pageLinks, ...(node.pages || [])],
     []
   );
-  pages.forEach(({ childMdx, name }) => {
-    if (name === 'releases') {
-      reporter.panicOnBuild(
-        '🚨  ERROR: file named "releases" is not allowed in src/content directory'
-      );
-      return;
-    }
-
+  pages.forEach(({ slug }) => {
+    const isOverviewPage = slug === '/releases';
     const matchingNavigationPage = navigationPages.find(
-      (page) =>
-        trimTrailingSlash(page.path) === trimTrailingSlash(childMdx.fields.slug)
+      (page) => trimTrailingSlash(page.path) === trimTrailingSlash(slug)
     );
     actions.createPage({
-      // This is the slug you created before
-      // (or `node.frontmatter.slug`)
-      path: childMdx.fields.slug,
+      path: slug,
       // This component will wrap our MDX content
-      component: require.resolve('./src/templates/page-content.js'),
-      // You can use the values in this context in
-      // our page layout component
+      component: isOverviewPage
+        ? require.resolve('./src/templates/release-notes-list.js')
```

Comment:
The release overview page needs to be rendered here now. It still has its own template but from a graphql type perspective is considered a "content page" and therefore is queried with all other content pages.

The file is now `src/content/releases.mdx` instead of `src/releases/index.mdx`, for the reasons mentioned above.